libs/base/Data/Contravariant.idr

```diff
@@ -0,0 +1,48 @@
+module Data.Contravariant
+
+infixl 4 >$, >$<, >$$<, $<
+
+||| Contravariant functors
+public export
+interface Contravariant (0 f : Type -> Type) where
+  contramap : (a -> b) -> f b -> f a
+
+  (>$) : b -> f b -> f a
+  v >$ fb = contramap (const v) fb
+
+||| If `f` is both `Functor` and `Contravariant` then by the time you factor in the
+||| laws of each of those classes, it can't actually use its argument in any
+||| meaningful capacity.
+public export %inline
+phantom : (Functor f, Contravariant f) => f a -> f b
+phantom fa = () >$ (fa $> ())
+
+||| This is an infix alias for `contramap`.
+public export %inline
+(>$<) : Contravariant f => (a -> b) -> f b -> f a
+(>$<) = contramap
+
+||| This is an infix version of `contramap` with the arguments flipped.
+public export %inline
+(>$$<) : Contravariant f => f b -> (a -> b) -> f a
+(>$$<) = flip contramap
```

Comment:
Shouldn't this be named `>&<`, since we have `<&>` in the Prelude for flipped `map` already?

Reply:
Ohh, I didn't even know about `<&>`. Good catch. Should be fixed now.